salt/modules/file.py

```diff
@@ -6091,7 +6120,11 @@ def manage_file(
                 return _error(ret, "Source file '{}' not found".format(source))
             # If the downloaded file came from a non salt server source verify
             # that it matches the intended sum value
-            if not skip_verify and urllib.parse.urlparse(source).scheme != "salt":
+            if (
+                not skip_verify
+                and urllib.parse.urlparse(source).scheme != "salt"
+                and not use_etag
+            ):
```

Comment:
**question** This same condition appears in three different places - is there opportunity for refactor here?

Reply:
Moved the check up earlier in the code to promote readability.